### ISSUE TITLE
Valid YAML line "something: 'Example: '" not parsed

### ISSUE THREAD

**Issue by jurruh** [smortex/yaml-sort#20]
Thanks for providing a ruby yaml-sorter. I experienced some strange behavior while using it.

Parsing this yaml:
```
---
something: 'Example: '
another: example
```

Raises this error:
```
Traceback (most recent call last):
        27: from /Users/jurre/.gem/ruby/2.7.5/bin/bundle:23:in `<main>'
        26: from /Users/jurre/.gem/ruby/2.7.5/bin/bundle:23:in `load'
        25: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/exe/bundle:36:in `<top (required)>'
        24: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/lib/bundler/friendly_errors.rb:120:in `with_friendly_errors'
        23: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/exe/bundle:48:in `block in <top (required)>'
        22: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/lib/bundler/cli.rb:25:in `start'
        21: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
        20: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/lib/bundler/cli.rb:31:in `dispatch'
        19: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
        18: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
        17: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
        16: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/lib/bundler/cli.rb:486:in `exec'
        15: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/lib/bundler/cli/exec.rb:23:in `run'
        14: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/lib/bundler/cli/exec.rb:58:in `kernel_load'
        13: from /Users/jurre/.gem/ruby/2.7.5/gems/bundler-2.3.24/lib/bundler/cli/exec.rb:58:in `load'
        12: from /Users/jurre/.gem/ruby/2.7.5/bin/yaml-sort:23:in `<top (required)>'
        11: from /Users/jurre/.gem/ruby/2.7.5/bin/yaml-sort:23:in `load'
        10: from /Users/jurre/.gem/ruby/2.7.5/gems/yaml-sort-2.2.0/exe/yaml-sort:7:in `<top (required)>'
         9: from /Users/jurre/.gem/ruby/2.7.5/gems/yaml-sort-2.2.0/lib/yaml/sort/cli.rb:46:in `execute'
         8: from /Users/jurre/.gem/ruby/2.7.5/gems/yaml-sort-2.2.0/lib/yaml/sort/cli.rb:46:in `each'
         7: from /Users/jurre/.gem/ruby/2.7.5/gems/yaml-sort-2.2.0/lib/yaml/sort/cli.rb:47:in `block in execute'
         6: from /Users/jurre/.gem/ruby/2.7.5/gems/yaml-sort-2.2.0/lib/yaml/sort/cli.rb:56:in `process_document'
         5: from /Users/jurre/.gem/ruby/2.7.5/gems/yaml-sort-2.2.0/lib/yaml/sort/cli.rb:72:in `sort_yaml'
         4: from parser.ra:191:in `parse'
         3: from parser.ra:60:in `scan'
         2: from parser.ra:60:in `loop'
         1: from parser.ra:61:in `block in scan'
```

**Comment by smortex** [comment]
Hey, thank you for the report and providing a failing sample. I can reproduce the issue and my version of Ruby seems more explicit about the failure:

```
parser.ra:80:in `+': no implicit conversion of nil into String (TypeError)
	from parser.ra:80:in `block in scan'
	from parser.ra:79:in `loop'
	from parser.ra:79:in `scan'
	from parser.ra:210:in `parse'
	from /usr/home/romain/Projects/yaml-sort/lib/yaml/sort/cli.rb:72:in `sort_yaml'
	from /usr/home/romain/Projects/yaml-sort/lib/yaml/sort/cli.rb:56:in `process_document'
	from /usr/home/romain/Projects/yaml-sort/lib/yaml/sort/cli.rb:47:in `block in execute'
	from /usr/home/romain/Projects/yaml-sort/lib/yaml/sort/cli.rb:46:in `each'
	from /usr/home/romain/Projects/yaml-sort/lib/yaml/sort/cli.rb:46:in `execute'
	from exe/yaml-sort:7:in `<top (required)>'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/lib/bundler/cli/exec.rb:58:in `load'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/lib/bundler/cli/exec.rb:58:in `kernel_load'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/lib/bundler/cli/exec.rb:23:in `run'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/lib/bundler/cli.rb:479:in `exec'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/lib/bundler/cli.rb:31:in `dispatch'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/lib/bundler/cli.rb:25:in `start'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/exe/bundle:48:in `block in <top (required)>'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
	from /usr/home/romain/.vendor/ruby/3.0/gems/bundler-2.3.16/exe/bundle:36:in `<top (required)>'
	from /usr/local/bin/bundle:25:in `load'
	from /usr/local/bin/bundle:25:in `<main>'
```

I am looking into this…

**Comment by smortex** [comment]
Fixed in yaml-sort-2.2.1, thanks!
